### PR TITLE
21 gtfs instance class diagram

### LINKED ISSUE #21: Gtfs_Instance class diagram

### PR DESCRIPTION
**Pull request by r-leyshon**
## Description

Class diagram for `GtfsInstance`

Fixes #21 

## Motivation and Context

Help with documentation & KO slide deck.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation

## How Has This Been Tested?

NA

## Checklist:

- [ ] My code follows the intended structure of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/gtfs/Gtfs_Instance_class_diagram.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      text-align: center;
+    }
+    </style>
+</header>
+
+```mermaid
+
+classDiagram
+    class GtfsInstance {
+        +feed : gtfs_kit.feed
```
Comment (SergioRec) [inline docs/gtfs/Gtfs_Instance_class_diagram.md:7]: Colon appears doubled in all lines except `feed` and `get_calendar_dates()` in readme preview.

File: docs/gtfs/Gtfs_Instance_class_diagram.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# GtfsInstance
+
+<header>
```
Comment (SergioRec) [inline docs/gtfs/Gtfs_Instance_class_diagram.md]: Header appears as raw html code in readme preview.

File: docs/gtfs/Gtfs_Instance_class_diagram.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# GtfsInstance
```
Comment (SergioRec) [inline docs/gtfs/Gtfs_Instance_class_diagram.md:1]: Can't render diagram either with Quarto or RStudio. Am I missing dependencies?

File: .gitignore [hunk 4]
```diff
@@ -308,6 +308,7 @@ vignettes/*.pdf
 # pkgdown site
 docs/*
 !docs/README.md
+!docs/gtfs
```
Comment (SergioRec) [inline .gitignore:311]: Quarto rendering will create an .html file, do we want to ignore it since the .md file is already there?

Comment (r-leyshon) [reply .gitignore:311]: Quarto does weird things with .mds. I discovered if you use an uppercase file extension `.MD`, quarto thinks it's an intermediate file in the quarto render process and then deletes it. Don't need the html file, but bit confused on why we're using quarto here.

File: docs/gtfs/Gtfs_Instance_class_diagram.md (hunk 3 above)
Comment (r-leyshon) [reply docs/gtfs/Gtfs_Instance_class_diagram.md:1]: Ah, I see. Apologies I've not intended for use with quarto and I should have made this more clear. GitHub-flavoured markdown will render the mermaid diagram, the way I intended for this to be assessed would be within the [GitHub UI](https://github.com/datasciencecampus/transport-network-performance/blob/21-gtfs_instance-class-diagram/docs/gtfs/Gtfs_Instance_class_diagram.md).

File: docs/gtfs/Gtfs_Instance_class_diagram.md (hunk 2 above)
Comment (r-leyshon) [reply docs/gtfs/Gtfs_Instance_class_diagram.md]: Yikes, yes it didn't render like that within my HTML preview in VSCode. I'll go ahead and move it to a separate stylesheet.

File: docs/gtfs/Gtfs_Instance_class_diagram.md (hunk 1 above)
Comment (r-leyshon) [reply docs/gtfs/Gtfs_Instance_class_diagram.md:7]: Good spot, not sure what's gone on there. There are some differences in the GitHub mermaid renderring to what I have in VSCode.

Comment (r-leyshon) [reply docs/gtfs/Gtfs_Instance_class_diagram.md:7]: Just to prove I'm not going mad:
image

Comment (r-leyshon) [reply docs/gtfs/Gtfs_Instance_class_diagram.md:7]: @SergioRec  This should be fixed now. Turns out that attributes need a colon but methods do not.

File: docs/gtfs/Gtfs_Instance_class_diagram.md (hunk 2 above)
Comment (r-leyshon) [reply docs/gtfs/Gtfs_Instance_class_diagram.md]: Removed header as styling not important and it seems GitHub markdown doesn't like to play ball with CSS.

**Review by SergioRec — APPROVED** [review]
Diagram renders correctly in Markdown preview.

**Pull request merged**